testinfra/utils/ansible_runner.py

```diff
@@ -69,28 +69,25 @@ def get_runner(cls, inventory):
     def fetch_inventory(self, host=None):
         '''Helper function for ansible-inventory'''
 
-        cmd = 'ansible-inventory -i %s' % self.host_list
+        cmd = [
+            'ansible-inventory',
+            '-i',
+            self.host_list
+        ]
         if host is not None:
-            cmd += ' --host=%s' % host
+            cmd.append('--host=%s' % host)
         else:
-            cmd += ' --list'
-
-        env = os.environ.copy()
-        env['ANSIBLE_NOCOLOR'] = "1"
-        p = subprocess.Popen(
-            cmd,
-            shell=True,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            env=env
-        )
-        (so, se) = p.communicate()
-
-        if p.returncode != 0:
-            msg = 'ansible-inventory failed: %s' % se
-            raise AnsibleInventoryException(msg)
-
-        inv = json.loads(so)
+            cmd.append('--list')
+
+        try:
+            so = subprocess.check_output(cmd)
+        except subprocess.CalledProcessError as e:
+            raise AnsibleInventoryException(e)
+
+        try:
+            inv = json.loads(so)
```

Comment:
This is failing with python3 as check_output returns bytes, which messes up json.loads()

```
E           TypeError: the JSON object must be str, not 'bytes'
```

https://github.com/ianw/testinfra/commit/821bf58cdbace9f7499cf184e200aeb46cabe705